### PR TITLE
pxe saltboot params

### PR DESCRIPTION
**Pull request by keichwa**
https://github.com/SUSE/spacewalk/issues/7450

### CODE REVIEW

File: modules/retail/pages/retail-formulas.adoc [hunk 1]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
 
 
+
+[[retail.sect.formulas.pxe.kernelparams]]
+=== Kernel Command Line Parameters
+Saltboot supports common kernel parameters and saltboot-specific kernel parameters.
+All the parameters can be entered in the [guimenu]``Kernel Command Line Parameters`` field of the PXE formula.
+
+[systemitem]``kiwidebug=1``::
+starts a shell on tty2 during boot and enables debug log level in Salt.
```
Comment (Loquacity) [inline modules/retail/pages/retail-formulas.adoc]:
```suggestion
Starts a shell on tty2 during boot and enables debug log level in Salt.
```

File: modules/retail/pages/retail-formulas.adoc [hunk 2]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
+
+[systemitem]``MASTER``::
+overrides the Salt master autodetection; for example:
```
Comment (Loquacity) [inline modules/retail/pages/retail-formulas.adoc]:
```suggestion
Overrides auto-detection of the Salt master. For example:
```

File: modules/retail/pages/retail-formulas.adoc [hunk 3]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
++
+----
+MASTER=myproxy.domain.com
+----
+
+[systemitem]``SALT_TIMEOUT``::
+fallback to the local boot if the Salt master does not apply the saltboot state within this timeout (default: 60s); for example:
```
Comment (Loquacity) [inline modules/retail/pages/retail-formulas.adoc]:
```suggestion
Fall back to the local boot if the Salt master does not apply the saltboot state within this timeout (default: 60 seconds). For example:
```

"Fall back" is a verb, "fallback" is an adjective.

File: modules/retail/pages/retail-formulas.adoc [hunk 4]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
+
+// see https://bugzilla.suse.com/show_bug.cgi?id=1111348#c3
+[systemitem]``DISABLE_HOSTNAME_ID`` and [systemitem]``DISABLE_UNIQUE_SUFFIX``::
+Option for manipulating the names of the terminals.
+
+[systemitem]``MINION_ID_PREFIX``::
+Branch ID set in the PXE formula form.
+
+[systemitem]``salt_device``
+device that contains the salt configuration.
```
Comment (Loquacity) [inline modules/retail/pages/retail-formulas.adoc]:
```suggestion
Device that contains the salt configuration.
```

File: modules/retail/pages/retail-formulas.adoc [hunk 5]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
+
+[systemitem]``root``
+device that contains the already deployed root filesystem.
```
Comment (Loquacity) [inline modules/retail/pages/retail-formulas.adoc]:
```suggestion
Device that contains the already deployed root filesystem.
```

File: modules/retail/pages/retail-formulas.adoc [hunk 6]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
+It is used for the fallback to local boot.
```
Comment (Loquacity) [inline modules/retail/pages/retail-formulas.adoc]:
```suggestion
Used for the fall back to local boot.
```

File: modules/retail/pages/retail-formulas.adoc (hunk 3 above)
Comment (keichwa) [reply modules/retail/pages/retail-formulas.adoc]: yes ;)  But in German we glue it together nevertheless.  But there are many other subtle differences :)

File: modules/retail/pages/retail-formulas.adoc (hunk 6 above)
Comment (keichwa) [reply modules/retail/pages/retail-formulas.adoc]: Hmm, @Loquacity .  Maybe, we'd better say in this case:

```Used for falling back to local boot.```

File: modules/retail/pages/retail-formulas.adoc [hunk 7]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
 
 
+
+[[retail.sect.formulas.pxe.kernelparams]]
+=== Kernel Command Line Parameters
+Saltboot supports common kernel parameters and saltboot-specific kernel parameters.
+All the parameters can be entered in the [guimenu]``Kernel Command Line Parameters`` field of the PXE formula.
+
+[systemitem]``kiwidebug=1``::
+Starts a shell on tty2 during boot and enables debug log level in Salt.
```
Comment (aaannz) [inline modules/retail/pages/retail-formulas.adoc]: Please add a warning that this should be used only in development environment and **must** not be used in production as it creates a major security hole.

File: modules/retail/pages/retail-formulas.adoc [hunk 8]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
+
+[systemitem]``MASTER``::
+Overrides auto-detection of the Salt master. For example:
++
+----
+MASTER=myproxy.domain.com
+----
+
+[systemitem]``SALT_TIMEOUT``::
+Fall back to the local boot if the Salt master does not apply the saltboot state within this timeout (default: 60 seconds). For example:
```
Comment (aaannz) [inline modules/retail/pages/retail-formulas.adoc]: Not to confuse customers, this timeout is always active and by default is 60s. This option can override this 60s default.
I would rephrase it like

`Override local boot fallback timeout if Salt master does not apply the saltboot state within this timeout (default: 60 seconds).`

File: modules/retail/pages/retail-formulas.adoc [hunk 9]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
++
+----
+SALT_TIMEOUT=300
+----
+
+// see https://bugzilla.suse.com/show_bug.cgi?id=1111348#c3
+[systemitem]``DISABLE_HOSTNAME_ID`` and [systemitem]``DISABLE_UNIQUE_SUFFIX``::
+Option for manipulating the names of the terminals.
```
Comment (aaannz) [inline modules/retail/pages/retail-formulas.adoc]: `DISABLE_UNIQUE_SUFFIX`

Setting this option to 1 disables adding random generated suffix to terminal minion id.

Please make sure your terminal has either unique hostname provided by dhcp and dns or terminal hardware comes with unique serial number stored in its SMBios memory.
Otherwise there is a risk of terminal minion id duplicity and failure to bootstrap the minion.

`DISABLE_HOSTNAME_ID`

If terminal has a hostname assigned by DHCP, it is by default used as a minion id. Setting this option to 1 disables this mechanism and SMBios information will be used as a minion id.

File: modules/retail/pages/retail-formulas.adoc [hunk 10]
```diff
@@ -230,14 +230,54 @@ The PXE formula is used to configure PXE booting on the branch server.
...
+
+[systemitem]``MINION_ID_PREFIX``::
```
Comment (aaannz) [inline modules/retail/pages/retail-formulas.adoc]: Please note that these three (`MINION_ID_PREFIX`, `salt_device`, `root`) are usually autoconfigured and should be used only in specific conditions, usually during debugging/development.

File: modules/retail/pages/retail-formulas.adoc (hunk 7 above)
Comment (keichwa) [reply modules/retail/pages/retail-formulas.adoc]: I'm going to add such a warning.

File: modules/retail/pages/retail-formulas.adoc (hunk 8 above)
Comment (keichwa) [reply modules/retail/pages/retail-formulas.adoc]: ok

File: modules/retail/pages/retail-formulas.adoc (hunk 9 above)
Comment (keichwa) [reply modules/retail/pages/retail-formulas.adoc]: ok

File: modules/retail/pages/retail-formulas.adoc (hunk 10 above)
Comment (keichwa) [reply modules/retail/pages/retail-formulas.adoc]: ok

**Comment by keichwa** [comment]
I think I submitted all proposed or requested changes.  Please review again, @aaannz and @Loquacity
